### PR TITLE
docs: Add ADR for module status metrics

### PR DESCRIPTION
**Pull request by chrkl**
## Description

Changes proposed in this pull request (what was done and why):

- Add ADR for module status metrics

Changes refer to particular issues, PRs or documents:

- #921 

## Traceability
- [ ] The PR is linked to a GitHub issue.
- [ ] New features have a milestone set.
- [ ] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [ ] The corresponding GitHub issue has a respective `area` and `kind` label.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [ ] Adjusted the documentation if the change is user-facing.
- [ ] The feature is unit-tested
- [ ] The feature is e2e-tested

### CODE REVIEW

**Review by shorim — DISMISSED** [review]

File: docs/contributor/arch/011-kymastatsmetrics.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      enabled: true
+    istio:
+      enabled: true
+  output:
+    otlp:
+      endpoint:
+        value: http://example.com:4317
+```
+
+Enabling the Kyma input will enable a custom metrics receiver, called `kymastats`, in the telemetry-metric-gateway that produces metrics for the module state and status conditions.
```
Comment (NHingerl) [inline docs/contributor/arch/011-kymastatsmetrics.md:53]:
```suggestion
Enabling the Kyma input will enable a custom metrics receiver, called `kymastats`, in the Telemetry metric gateway that produces metrics for the module state and status conditions.
```

File: docs/contributor/arch/011-kymastatsmetrics.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+The receiver configuration will follow the shown example.
```
Comment (NHingerl) [inline docs/contributor/arch/011-kymastatsmetrics.md:54]: What does this mean? Do you plan to add more content?

File: docs/contributor/arch/011-kymastatsmetrics.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+The receiver needs the following properties:
+
+- **auth_type**: The way to authenticate with the Kubernetes API. Possible values are `serviceAccount` (default) or `kubeConfig`.
+- **collection_interval**: The interval that is used to emit metrics.
+- **api_groups**: List of API groups to scrape. Most of the Kyma modules should use the `operator.kyma-project.io` API group. The list can be extended to support monitoring custom modules. Every CRD in the listed groups is assumed to represent a module.
+
+### Custom Metrics Receiver for OpenTelemetry Collector
+
+We assume the status subresource of a module CRD to contain a `conditions` list that uses the [meta/v1/Condition](https://pkg.go.dev/k8s.io/apimachinery@v0.30.0/pkg/apis/meta/v1#Condition) type and an overarching state attribute. We assume positive polarity for all conditions.
```
Comment (NHingerl) [inline docs/contributor/arch/011-kymastatsmetrics.md:74]:
```suggestion
We assume that the status subresource of a module CRD contains a `conditions` list that uses the type [meta/v1/Condition](https://pkg.go.dev/k8s.io/apimachinery@v0.30.0/pkg/apis/meta/v1#Condition), and an overarching state attribute. We assume positive polarity for all conditions.
```

File: docs/contributor/arch/011-kymastatsmetrics.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+An example for this structure is the status subresource of the Telemetry module:
```
Comment (NHingerl) [inline docs/contributor/arch/011-kymastatsmetrics.md:76]: I assume we should format it as [parameter](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/03-formatting.md#use-code-font-for-these-items)? 

```suggestion
As an example for this structure, see the **status** subresource of the Telemetry module:
```

File: docs/contributor/arch/011-kymastatsmetrics.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+Proposed
+
+## Context
+
+The [Advanced pipeline status based on data flow](https://github.com/kyma-project/telemetry-manager/issues/425) epic describes efforts to make problems of the Telemetry module more transparent to users by utilizing CRD status conditions.
+To ease day-two operations, the status of the Telemetry and other Kyma modules should by available as metrics to enable users to integrate them into their monitoring system. For instance, by setting up alerts for a module status that differs from the expected value.
+
+The [kube-state-metrics](https://github.com/kubernetes/kube-state-metrics) exporter provides a way to [export metrics for custom resources](https://github.com/kubernetes/kube-state-metrics/blob/main/docs/metrics/extend/customresourcestate-metrics.md).
+This record describes a way to integrate similar functionality to the Telemetry module's OpenTelemetry Collectors. The solution should avoid the maintenance overhead of an additional third-party-image and allow a dynamic configuration, based on the active Kyma modules.
```
[the remark this answers is not in the document]
Comment (chrkl) [reply docs/contributor/arch/011-kymastatsmetrics.md]: This ADR (architecture decision record). I changed it to "this document" to make the meaning more obvious.

File: docs/contributor/arch/011-kymastatsmetrics.md (hunk 4 above)
Comment (chrkl) [reply docs/contributor/arch/011-kymastatsmetrics.md]: The shown yaml is generated by the operator. So rather not a parameter.

File: docs/contributor/arch/011-kymastatsmetrics.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (NHingerl) [reply docs/contributor/arch/011-kymastatsmetrics.md]: Aaaah! Thank you :)

File: docs/contributor/arch/011-kymastatsmetrics.md (hunk 4 above)
Comment (NHingerl) [reply docs/contributor/arch/011-kymastatsmetrics.md]: Then is it a [code example](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/03-formatting.md#items-written-in-code-font)?

```suggestion
As an example for this structure, see the `status` subresource of the Telemetry module:
```

Comment (chrkl) [reply docs/contributor/arch/011-kymastatsmetrics.md]: I think that fits better according to the table in the guide. Changed it.

File: docs/contributor/arch/011-kymastatsmetrics.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+      enabled: true
+  output:
+    otlp:
+      endpoint:
+        value: http://example.com:4317
+```
+
+Enabling the Kyma input will enable a custom metrics receiver, called `kymastats`, in the Telemetry metric gateway that produces metrics for the module state and status conditions.
+
+The receiver configuration will follow the shown example:
```
Comment (NHingerl) [inline docs/contributor/arch/011-kymastatsmetrics.md:57]:
```suggestion
See the following example for the receiver configuration:
```

File: docs/contributor/arch/011-kymastatsmetrics.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+receivers:
+  kymastats:
+    k8s_cluster:
+        auth_type: serviceAccount
+    collection_interval: 30s
+    api_groups:
+    - operator.kyma-project.io
+```
+
+The receiver needs the following properties:
```
Comment (NHingerl) [inline docs/contributor/arch/011-kymastatsmetrics.md:69]: I'd put this section (line 69-73) _before_ the example of line 57-67.

File: docs/contributor/arch/011-kymastatsmetrics.md [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+
+- The OpenTelemetry Collector provides interfaces for enhancements by implementing custom receiver plugins. The OpenTelemetry project provides [documentation](https://opentelemetry.io/docs/collector/building/receiver/) on implementing a custom receiver and adding it to a custom distribution of the Collector.
+
+## Decision
+
+Due to the restrictions of available telemetry resources for Kubernetes resources, building a custom receiver is the most suitable option.
+
+### Extended MetricPipeline API
+
+To activate module status metrics as a new input, the MetricPipeline CRD needs a _module list_. If the module list is empty, metrics of all active modules are collected. As in the following example, users can select the Kyma modules of interest:
```
Comment (NHingerl) [inline docs/contributor/arch/011-kymastatsmetrics.md:30]:
```suggestion
To activate module status metrics as a new input, the MetricPipeline CRD needs a _module list_. If the module list is empty, metrics of all active modules are collected. Users can select the Kyma modules of interest; in the following example, only the `telemetry` module is selected:
```

File: docs/contributor/arch/011-kymastatsmetrics.md (hunk 2 above)
Comment (NHingerl) [reply docs/contributor/arch/011-kymastatsmetrics.md]: Thanks! I made a new suggestion :)

File: docs/contributor/arch/011-kymastatsmetrics.md (hunk 7 above)
Comment (chrkl) [reply docs/contributor/arch/011-kymastatsmetrics.md]: Done

**Review by NHingerl — APPROVED** [review]

**Pull request merged**